tests/test_config.py

```diff
@@ -437,6 +474,29 @@ def test_charmhub_frozen():
         config.api_url = "broken"
 
 
+def test_charmhub_underscore_backwards_compatibility(create_config, tmp_path, caplog):
+    """Support underscore in these attributes for a while."""
+    caplog.set_level(logging.WARNING, logger="charmcraft")
+
+    create_config(
+        """
+        type: charm  # mandatory
+        charmhub:
+            storage_url: https://server1.com
+            api_url: https://server2.com
+            registry_url: https://server3.com
+    """
+    )
+    cfg = load(tmp_path)
+    assert cfg.charmhub.storage_url == "https://server1.com"
+    assert cfg.charmhub.api_url == "https://server2.com"
+    assert cfg.charmhub.registry_url == "https://server3.com"
+    deprecation_msg = (
+        "DEPRECATED: Configuration keywords are now separated using dashes."
+    )
+    assert deprecation_msg in [rec.message for rec in caplog.records]
```

Comment:
One thing we have as tech debt in Snapcraft is to log the deprecation notice only once. It would be nice, but not mandatory to have that here from the get go.

Reply:
Good idea, added this detail.